### ISSUE TITLE
Errors message occurs when running Timeseries_Sample.ipynb

### ISSUE THREAD

**Issue by corinne-hcr** [e-mission/e-mission-docs#601]
There is an error message after running this line
```test_user_id = all_users.iloc[1].uuid # replace with UUID from above```
```IndexError: single positional indexer is out-of-bounds```

1. I searched ```.iloc ``` and the error ```IndexError: single positional indexer is out-of-bounds```
2. I found a suggestion online that shows ```.iloc```is for the position, and ```.loc``` is for the index
3. As for  "all_user" in the code```all_users = pd.DataFrame(list(edb.get_uuid_db().find({}, {"user_email":1, "uuid": 1, "_id": 0})))```
I didn't found ```"user_email"``` ```"uuid"``` in ```get_database.py```

4. After running ```docker run -p 27017:27017 mongo:4.4.0```, I saw the data like this ```{"t":{"$date":"2021-01-14T00:25:40.755+00:00"},"s":"I",  "c":"CONTROL",  "id":23285,   "ctx":"main","msg":"Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'"}```I think I should connect the MongoDB successfully. 

Now, I don't know what to do further.

**Comment by shankari** [comment]
4. Yes, you appear to have connected to mongodb successfully
3. Not sure why you would expect to find `user_email` and `uuid` in `get_database.py`
2. yes, `iloc` is for the position. Do you know that there is an entry at position 1?

**Comment by corinne-hcr** [comment]
For 2, could you tell me what you mean by an entry at position 1?

**Comment by shankari** [comment]
As you probably saw from the documentation, `all_users.iloc[1]` is reading the entry at position 1 in all_users.
Does all_users have this entry?

**Comment by corinne-hcr** [comment]
Do you mean I should look at here ```all_users = pd.DataFrame(list(edb.get_uuid_db().find({}, {"user_email":1, "uuid": 1, "_id": 0})))``` and ```{"user_email":1, "uuid": 1, "_id": 0}``` is 1 entry?

**Comment by shankari** [comment]
`.find` is a pymongo command - you can look up its parameters to understand more.

**Comment by corinne-hcr** [comment]
For ```edb.get_uuid_db()``` I found ```UUIDs = _get_current_db().Stage_uuids``` But I don't understand```Stage_uuids```

**Comment by shankari** [comment]
did you see what `_get_current_db()` returns? Maybe type out `edb.get_uuid_db()` in a cell and see what it returns?

**Comment by shankari** [comment]
Reported by @corinne-hcr  in private chat: @corinne-hcr, please put technical updates into the issue in the future.

I still don't fully understand the line. 

```
edb.get_uuid_db()
```

will return a collection, though I can't figure out what ```Stage_database``` and ```Stage_uuids``` are. But it becomes

```
collection.find()
```

and should return all documents with the keys:

```
"user_email"
```

and

```
"uuid"
```

`"_id"` is filtered out 

```
list()
```

makes all the keys into a list.
Then, I don't know why it would raise an IndexError.

**Comment by shankari** [comment]
>  Then, I don't know why it would raise an IndexError.

It raises an index error because the size of the list is too small.
If you run the command in a separate cell, you can confirm this by printing out the list.

Have you loaded any data into the system?

**Comment by corinne-hcr** [comment]
I am not sure if I have loaded any data. Since I cloned the e-mission-server, I thought the data was in the folders. But when I run the line separately, the result is empty. It seems I don't have the data.

1. When I run```edb.get_uuid_db()```
The result is ```Collection(Database(MongoClient(host=['localhost:27017'], document_class=dict, tz_aware=False, connect=True), 'Stage_database'), 'Stage_uuids')```

2. I run ```edb.get_uuid_db().find({}, {"user_email":1, "uuid": 1, "_id": 0})```
The result is ```<pymongo.cursor.Cursor at 0x7f80934b4750>```

3. I run``` list(edb.get_uuid_db().find({}, {"user_email":1, "uuid": 1, "_id": 0})) ```
The result is ```[]```

**Comment by shankari** [comment]
https://github.com/e-mission/e-mission-server/#loading-test-data

**Comment by corinne-hcr** [comment]
I am now at these two steps:

- Setup the ```emission``` environment
- Verify that you are in the right environment 

I have verified my environment.
```(emission) ...$ conda env list```
```# conda environments:```
One of the environment is:
```emission              *  /Applications/anaconda3/envs/emission```

My question is, since there are ```.ipynb```and```.py```files, should I also use ```$ source setup/setup_notebook.sh```to set up the environment?

**Comment by shankari** [comment]
ah good question. I haven't used that for a while, but let me check what it has.

**Comment by shankari** [comment]
As we can see by diffing `environment36.yml` and `environment36.notebook.yml`, or by checking `setup/environment36.notebook.additions.yml`, there are 4 new packages for notebooks.

Depending on your notebook equivalent, you may already have a couple of the packages, but probably need to install `branca` and `folium`.

**Comment by corinne-hcr** [comment]
I am using Anaconda-Navigator now. There is a JupyterLab, I use it to run ```Timeseries_Sample.ipynb```

**Comment by shankari** [comment]
how did you do that and have `pymongo` work when you didn't have the `emission` environment set up?

**Comment by corinne-hcr** [comment]
I can set environments through Anaconda-Navigator, and I can search for packages there. But I installed the ```pymongo``` in the base(root).

**Comment by shankari** [comment]
ok, so you should switch to the e-mission environment in any place where you want to run e-mission code. That's how you ensure that you have all the dependencies.

wrt `setup_notebook`, I would run it as well to ensure that the viz libraries (branca and folium) are in the environment.

**Comment by corinne-hcr** [comment]
After ```setup_notebook```, I see the prompt
```#
# To activate this environment, use
#
#     $ conda activate emission
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```


Then, I run this
```(emission) ChunruideMacBook-Pro:e-mission-server corinne$ conda activate emission```

But does it affect the cmd line in the instruction under ```Run```?

> **Run**
> 
> 1. Activate the emission conda environment
> ```source setup/activate.sh```
> 

When I run ```$ source setup/activate.sh```, the prompt is ```-bash: /Users/corinne/miniconda-4.8.3/etc/profile.d/conda.sh: No such file or directory```

I also try

> 2. For OSX manual installs, start the database (Note: mongodb appears to be installed as a service on Windows devices and it starts automatically on reboot).
```$ mongod```

The prompt is 
```(emission) ChunruideMacBook-Pro:e-mission-server corinne$ mongod```
```-bash: mongod: command not found```

**Comment by shankari** [comment]
This is only for OSX **manual installs**. As we discussed today, you are using the docker install so this doesn't apply to you. Again, you only need to run the commands that are specific to your choices.

```
(emission) ChunruideMacBook-Pro:e-mission-server corinne$ mongod
-bash: mongod: command not found
```

**Comment by shankari** [comment]
if you look at the source code for `source setup/activate.sh`, it calls 

```
source setup/activate_conda.sh
conda activate emission
```

you can try running each of them separately and seeing which fails.

Or you can look at the error says that the file `/Users/corinne/miniconda-4.8.3/etc/profile.d/conda.sh` is not found.
It looks like you don't have that file.
Did you install `miniconda`?

**Comment by corinne-hcr** [comment]
Oh, I see. Then it looks like I already activated the conda environment.

As for ```minicoda```,I have already run this:
```(base) ChunruideMacBook-Pro:e-mission-server corinne$ source setup/setup_conda.sh```

The result is:
```
Installing for version 4.8.3 and platform 
Usage: setup_conda.sh <platform>
   Assumes that the EXP_CONDA_VER variable is set
   Platform options are Linux-x86_64, MacOSX-x86_64
   For Windows, manually download and install https://repo.anaconda.com/miniconda/Miniconda3-4.8.3-Windows-x86_64.exe
```

**Comment by corinne-hcr** [comment]
Now I go to 

> 4. Start the server
> ```$ ./e-mission-py.bash emission/net/api/cfc_webapp.py```

and 

> 5. Test your connection to the server
> 
> - Using a web browser, go to http://localhost:8080 

The website shows some labels ```Home``` ```Analysis``` ``` Metrics``` ```Trip Planning``` ```Game```

Is the connection successful? 
Next step is ```Loading test data```?

**Comment by shankari** [comment]
This is telling you to run `setup_conda.sh` with a platform option.

```
Usage: setup_conda.sh <platform>
```

If you haven't done that, you haven't installed miniconda.
It looks like `conda activate` is using your UI-based conda install.

It looks like several of your recent questions are because you are not following the standard setup.

Yes, the connection is successful and you should load test data.

**Comment by corinne-hcr** [comment]
I'm sorry I didn't post the previous steps since I thought I ran it correctly. 
Now, should I back to ```Install miniconda```?

**Comment by shankari** [comment]
in the interests of time, you can probably go on now, but if you run into other problems that you don't know how to resolve, you may want to try installing miniconda first.

**Comment by corinne-hcr** [comment]
I try to install miniconda. The prompt is

> Successfully installed at /Users/corinne/miniconda-4.8.3. Please activate with 'source setup/activateXXX.sh' in every terminal where you want to use conda

In ```source setup/activateXXX.sh```, what is ```XXX```?

**Comment by shankari** [comment]
> I also upgrade conda version to 4.9.2, since there is a warning:

Don't. The install uses 4.8.3, and all the automated testing uses 4.8.3. 4.9.2 should also work, but I haven't tested it, and I can't help you if it breaks.

> In source setup/activateXXX.sh, what is XXX?

XXX is a placeholder. There are multiple activate scripts. You pick the appropriate one - e.g. `activate.sh` or `activate_tests.sh`....

**Comment by corinne-hcr** [comment]
But I've upgraded the version, how can I delete the latest one in order to re-install 4.8.3?

**Comment by shankari** [comment]
`setup/teardown_conda.sh`

In general, I don't have documentation for every script because there is no dedicated documentation team and I don't have time. You should look around in the directories and try out scripts that look like they will do what you want.

**Comment by corinne-hcr** [comment]
But after I use 
```(base) ChunruideMacBook-Pro:e-mission-server corinne$ setup/teardown_conda.sh```
It shows ```-bash: setup/teardown_conda.sh: Permission denied```

**Comment by shankari** [comment]
that is the script. To run the script, you have to use `source setup/teardown_conda.sh` like you run the other files in setup.

**Comment by corinne-hcr** [comment]
But it seems doesn't work
```(base) ChunruideMacBook-Pro:e-mission-server corinne$ source setup/teardown_conda.sh```
```(base) ChunruideMacBook-Pro:e-mission-server corinne$ conda --version```
```conda 4.9.2```

**Comment by shankari** [comment]
I suspect that is the desktop version of conda that you already installed with Anaconda Navigator.
You can use

```
which conda
```

to confirm

**Comment by corinne-hcr** [comment]
Oh, you are right!!

> (base) ChunruideMacBook-Pro:e-mission-server corinne$ which conda
> /Applications/anaconda3/bin/conda

**Comment by shankari** [comment]
The setup script installs particular versions of conda and all the packages. There is automated testing to ensure that they are still valid. If you want to use any other versions, you are on your own 😇

**Comment by corinne-hcr** [comment]
I check the conda info

```(base) ChunruideMacBook-Pro:e-mission-server corinne$ conda info```

     

> active environment : /Applications/anaconda3
>     active env location : /Applications/anaconda3
>             shell level : 1
>        user config file : /Users/corinne/.condarc
>  populated config files : /Users/corinne/.condarc
>           conda version : 4.9.2
>     conda-build version : not installed
>          python version : 3.8.3.final.0
>        virtual packages : __osx=10.15.7=0
>                           __unix=0=0
>                           __archspec=1=x86_64
>        base environment : /Users/corinne/miniconda-4.8.3  (writable)
>            channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
>                           https://repo.anaconda.com/pkgs/main/noarch
>                           https://repo.anaconda.com/pkgs/r/osx-64
>                           https://repo.anaconda.com/pkgs/r/noarch
>           package cache : /Users/corinne/miniconda-4.8.3/pkgs
>                           /Users/corinne/.conda/pkgs
>        envs directories : /Users/corinne/miniconda-4.8.3/envs
>                           /Users/corinne/.conda/envs
>                platform : osx-64
>              user-agent : conda/4.9.2 requests/2.25.1 CPython/3.8.3 Darwin/19.6.0 OSX/10.15.7
>                 UID:GID : 501:20
>              netrc file : None
>            offline mode : False

**Comment by shankari** [comment]
ok... not sure what your question is?!

**Comment by corinne-hcr** [comment]
So, the ```miniconda-4.8.3```is still there right? Should I set up ```emission``` environment?

**Comment by shankari** [comment]
I don't see that miniconda-4.8.3 is still there - the conda that you are getting the info from is `/Applications/anaconda3`

Instead of my trying to debug your custom install, which will never be replicated, can you uninstall all other anacondas (or remove them from your path) and start again?

Follow the setup instructions exactly, starting from a clean slate. 
You can see what is expected to happen from the automated testing.

https://github.com/e-mission/e-mission-server/runs/1675692604?check_suite_focus=true

The automated testing runs the same commands as the setup. As you run them, you should see the same answers.
If you do, then everything will work.

**Comment by shankari** [comment]
Remember not to run the teardown scripts!

**Comment by corinne-hcr** [comment]
OK. Just want to make sure.
Should I Enter ```rm -rf ~/anaconda3``` to remove the directory
or should I Full uninstall using ```anaconda-Clean```?

**Comment by shankari** [comment]
I have not worked with anaconda desktop, so I don't have any special insights.
I would follow the instructions on their website, and follow them exactly.

Also note that you can see the exact commands (in light blue) on the automated test logs as well.

<img width="1257" alt="Screen Shot 2021-01-14 at 9 49 43 PM" src="https://user-images.githubusercontent.com/2423263/104686709-9a6e0100-56b2-11eb-8197-b3c1d1e0f6e8.png">

**Comment by corinne-hcr** [comment]
Here is what I have now
```(base) ChunruideMacBook-Pro:~ corinne$ conda info --envs```
```
# conda environments:
#
base                  *  /Applications/anaconda3
emission                 /Applications/anaconda3/envs/emission
opencv                   /Applications/anaconda3/envs/opencv
                         /Users/corinne/miniconda-4.8.3
```

This is the conda version
```(base) ChunruideMacBook-Pro:~ corinne$ conda -V```
```conda 4.8.3
```

I want to remove the ```emission                 /Applications/anaconda3/envs/emission``` environment because it would upgrade my conda version.

Also, I want to remove  ```/Users/corinne/miniconda-4.8.3``` because it would stop me from re-installing miniconda. I am not sure whether this environment works fine. I prefer re-install it.

So, should I go to the e-mission-server directory and directly run ```$ source setup/teardown.sh```?

**Comment by shankari** [comment]
what does `which conda` show?

**Comment by corinne-hcr** [comment]
```(base) ChunruideMacBook-Pro:~ corinne$ which conda```
```/Applications/anaconda3/bin/conda```

**Comment by shankari** [comment]
so that is still your original, manually installed conda. you have two options:
- delete the manual install of anaconda and use the script which will install miniconda, or
- use the existing install of anaconda, don't install miniconda and just use the emission environment

Is your `emission` environment is broken in some way?

**Comment by shankari** [comment]
I don't even think you need to remove your manual install. Just put the miniconda first in your PATH.

**Comment by shankari** [comment]
> So, should I go to the e-mission-server directory and directly run $ source setup/teardown.sh?

This will remove the `miniconda` environment and leave the one in /`Applications`
At this point, the easiest option might be to check `.bash_profile` and `.bashrc`, find the place where the `/Applications` one is in the path and remove it.
Open a new terminal, check that `which conda` doesn't return anything and move along

**Comment by corinne-hcr** [comment]
After I use ```vim -r .bash_profile``` It shows
```
Using swap file ".bash_profile.swp"
Original file "~/.bash_profile"
Recovery completed. You should check if everything is OK.
(You might want to write out this file under another name
and run diff with the original file to check for changes)
You may want to delete the .swp file now.
```

**Comment by corinne-hcr** [comment]
Here is what it shows after pressing enter after using ```vim -r .bash_profile```
```
Last login: Fri Jan 15 10:00:08 on ttys000

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
(base) ChunruideMacBook-Pro:~ corinne$ pwd
/Users/corinne
(base) ChunruideMacBook-Pro:~ corinne$ which conda
/Applications/anaconda3/bin/conda
(base) ChunruideMacBook-Pro:~ corinne$ conda -V
conda 4.8.3
(base) ChunruideMacBook-Pro:~ corinne$ ls -ltra
total 320
drwxr-xr-x+   4 corinne  staff    128 Aug 27  2018 Public
drwx------    3 corinne  staff     96 Sep  4  2018 .cups
-rw-r--r--    1 corinne  staff  81413 May  7  2019 java_error_in_pycharm_298.log
drwxr-xr-x    3 corinne  staff     96 Aug 28  2019 .tooling
drwxr-xr-x    6 root     admin    192 Oct 12  2019 ..
drwx------+   5 corinne  staff    160 Oct 17  2019 Music
drwxr-xr-x    4 corinne  staff    128 Nov  7  2019 .vscode
drwx------@  78 corinne  staff   2496 Nov 23  2019 Library
-rw-r--r--    1 corinne  staff   1862 Jan  4  2020 .bash_profile-anaconda3.bak
drwx------    4 corinne  staff    128 Jan  6  2020 .config
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 hdf5==1.10.4=hfa1e0ec_0
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 h5py==2.9.0=py27h3134771_0
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 hdf5[version=>=1.10.4,<1.10.5.0a0]
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 libopencv==3.4.2=h0d3447a_1
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 hdf5[version=>=1.10.2,<1.10.3.0a0|>=1.8.18,<1.8.19.0a0|>=1.8.20,<1.9.0a0]
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 hdf5[version=>=1.10.1,<1.10.2.0a0|>=1.10.2,<1.10.3.0a0|>=1.10.4,<1.10.5.0a0|>=1.8.18,<1.8.19.0a0|>=1.8.20,<1.9.0a0]
-rw-------    1 corinne  staff     98 Jan  7  2020 .python_history
drwx------    4 corinne  staff    128 Feb  5  2020 .putty
drwx------    4 corinne  staff    128 Jul  3  2020 .local
drwxr-xr-x  468 corinne  staff  14976 Jul 12  2020 git
-rw-r--r--    1 corinne  staff     53 Jul 12  2020 .gitconfig
-rw-r--r--    1 corinne  staff   2461 Jul 13  2020 .bash_profile
drwx------    5 corinne  staff    160 Jul 13  2020 .ssh
-r--------    1 corinne  staff      9 Aug 25 18:15 .CFUserTextEncoding
drwxr-xr-x    7 corinne  staff    224 Oct  4 18:22 Calibre Library
drwx------@  10 corinne  staff    320 Oct  5 13:09 Documents
drwxr-xr-x    4 corinne  staff    128 Dec 20 10:56 Applications
drwx------+   7 corinne  staff    224 Dec 22 13:15 Movies
drwx------  161 corinne  staff   5152 Jan 11 01:43 .Trash
drwx------+  68 corinne  staff   2176 Jan 12 07:40 Pictures
drwxr-xr-x   17 corinne  staff    544 Jan 12 15:09 getting-started
drwx------@ 185 corinne  staff   5920 Jan 14 17:13 Downloads
drwxr-xr-x    6 corinne  staff    192 Jan 15 01:08 .anaconda_backup
drwxrwsr-x    3 corinne  staff     96 Jan 15 02:26 .conda
-rw-r--r--    1 corinne  staff     17 Jan 15 03:19 .condarc
drwxr-xr-x   15 corinne  staff    480 Jan 15 03:41 miniconda-4.8.3
drwx------@   4 corinne  staff    128 Jan 15 09:58 Creative Cloud Files
drwxr-xr-x   11 corinne  staff    352 Jan 15 09:58 .docker
drwx------@ 109 corinne  staff   3488 Jan 15 13:29 Desktop
-rw-r--r--    1 corinne  staff  12288 Jan 15 15:19 .bash_profile.swp
-rw-------    1 corinne  staff  10861 Jan 15 15:19 .viminfo
drwxr-xr-x+  45 corinne  staff   1440 Jan 15 15:19 .
-rw-------    1 corinne  staff  13956 Jan 15 15:19 .bash_history
drwx------   47 corinne  staff   1504 Jan 15 15:19 .bash_sessions
-rw-r--r--@   1 corinne  staff  14340 Jan 15 15:25 .DS_Store
(base) ChunruideMacBook-Pro:~ corinne$ env
TERM_PROGRAM=Apple_Terminal
SHELL=/bin/bash
TERM=xterm-256color
TMPDIR=/var/folders/8g/ppzqqrfn5z3_w8zjqykgs9fh0000gn/T/
CONDA_SHLVL=1
CONDA_PROMPT_MODIFIER=(base) 
TERM_PROGRAM_VERSION=433
TERM_SESSION_ID=90B29B78-432E-4747-AB57-D1B536D3FC24
USER=corinne
CONDA_EXE=/Applications/anaconda3/bin/conda
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.NFYubjRlMz/Listeners
_CE_CONDA=
PATH=/Applications/anaconda3/bin:/Applications/anaconda3/condabin:/anaconda3/bin:/Library/Frameworks/Python.framework/Versions/3.7/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin
LaunchInstanceID=AD966052-39BE-44E0-9B04-8022A86D57E9
CONDA_PREFIX=/Applications/anaconda3
PWD=/Users/corinne
XPC_FLAGS=0x0
_CE_M=
XPC_SERVICE_NAME=0
SHLVL=1
HOME=/Users/corinne
CONDA_PYTHON_EXE=/Applications/anaconda3/bin/python
LOGNAME=corinne
LC_CTYPE=UTF-8
CONDA_DEFAULT_ENV=base
SECURITYSESSIONID=186a6
_=/usr/bin/env
(base) ChunruideMacBook-Pro:~ corinne$ cd desktop
(base) ChunruideMacBook-Pro:desktop corinne$ cd e-mission
(base) ChunruideMacBook-Pro:e-mission corinne$ cd
(base) ChunruideMacBook-Pro:~ corinne$ cd ~/e-mission-server
-bash: cd: /Users/corinne/e-mission-server: No such file or directory
(base) ChunruideMacBook-Pro:~ corinne$ cd desktop
(base) ChunruideMacBook-Pro:desktop corinne$ cd e-mission
(base) ChunruideMacBook-Pro:e-mission corinne$ cd e-mission-server
(base) ChunruideMacBook-Pro:e-mission-server corinne$ source setup/setup.sh
For conda, found 4.8.3, expected 4.8.3, all is good!
Installing using conda now
Collecting package metadata (repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.8.3
  latest version: 4.9.2

Please update conda by running

    $ conda update -n base -c defaults conda


Ran pip subprocess with arguments:
['/Applications/anaconda3/envs/emission/bin/python', '-m', 'pip', 'install', '-U', '-r', '/Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt']
Pip subprocess output:
Requirement already up-to-date: jwcrypto==0.7 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from -r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 1)) (0.7)
Requirement already up-to-date: pyfcm==1.4.7 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from -r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 2)) (1.4.7)
Requirement already up-to-date: pygeocoder==1.2.5 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from -r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 3)) (1.2.5)
Requirement already up-to-date: pymongo==3.11.0 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from -r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 4)) (3.11.0)
Requirement already satisfied, skipping upgrade: cryptography>=1.5 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from jwcrypto==0.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 1)) (3.3.1)
Requirement already satisfied, skipping upgrade: requests in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from pyfcm==1.4.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 2)) (2.24.0)
Requirement already satisfied, skipping upgrade: six>=1.4.1 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from cryptography>=1.5->jwcrypto==0.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 1)) (1.15.0)
Requirement already satisfied, skipping upgrade: cffi>=1.12 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from cryptography>=1.5->jwcrypto==0.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 1)) (1.14.4)
Requirement already satisfied, skipping upgrade: chardet<4,>=3.0.2 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from requests->pyfcm==1.4.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 2)) (3.0.4)
Requirement already satisfied, skipping upgrade: idna<3,>=2.5 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from requests->pyfcm==1.4.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 2)) (2.10)
Requirement already satisfied, skipping upgrade: certifi>=2017.4.17 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from requests->pyfcm==1.4.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 2)) (2020.12.5)
Requirement already satisfied, skipping upgrade: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from requests->pyfcm==1.4.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 2)) (1.25.11)
Requirement already satisfied, skipping upgrade: pycparser in /Applications/anaconda3/envs/emission/lib/python3.7/site-packages (from cffi>=1.12->cryptography>=1.5->jwcrypto==0.7->-r /Users/corinne/Desktop/e-mission/e-mission-server/setup/condaenv._e8senrr.requirements.txt (line 1)) (2.20)

#
# To activate this environment, use
#
#     $ conda activate emission
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(base) ChunruideMacBook-Pro:e-mission-server corinne$ conda env
usage: conda-env [-h] {create,export,list,remove,update,config} ...

positional arguments:
  {create,export,list,remove,update,config}
    create              Create an environment based on an environment file
    export              Export a given environment
    list                List the Conda environments
    remove              Remove an environment
    update              Update the current environment based on environment
                        file
    config              Configure a conda environment

optional arguments:
  -h, --help            Show this help message and exit.

conda commands available from other packages:
  build
  convert
  debug
  develop
  env
  index
  inspect
  metapackage
  render
  server
  skeleton
  verify
(base) ChunruideMacBook-Pro:e-mission-server corinne$ cd
(base) ChunruideMacBook-Pro:~ corinne$ conda info

     active environment : base
    active env location : /Applications/anaconda3
            shell level : 1
       user config file : /Users/corinne/.condarc
 populated config files : /Users/corinne/.condarc
          conda version : 4.8.3
    conda-build version : 3.18.9
         python version : 3.7.4.final.0
       virtual packages : __osx=10.15.7
       base environment : /Applications/anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Applications/anaconda3/pkgs
                          /Users/corinne/.conda/pkgs
       envs directories : /Applications/anaconda3/envs
                          /Users/corinne/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.3 requests/2.22.0 CPython/3.7.4 Darwin/19.6.0 OSX/10.15.7
                UID:GID : 501:20
             netrc file : None
           offline mode : False

(base) ChunruideMacBook-Pro:~ corinne$ conda env list
# conda environments:
#
base                  *  /Applications/anaconda3
emission                 /Applications/anaconda3/envs/emission
opencv                   /Applications/anaconda3/envs/opencv
                         /Users/corinne/miniconda-4.8.3

(base) ChunruideMacBook-Pro:~ corinne$ ls -ltra
total 320
drwxr-xr-x+   4 corinne  staff    128 Aug 27  2018 Public
drwx------    3 corinne  staff     96 Sep  4  2018 .cups
-rw-r--r--    1 corinne  staff  81413 May  7  2019 java_error_in_pycharm_298.log
drwxr-xr-x    3 corinne  staff     96 Aug 28  2019 .tooling
drwxr-xr-x    6 root     admin    192 Oct 12  2019 ..
drwx------+   5 corinne  staff    160 Oct 17  2019 Music
drwxr-xr-x    4 corinne  staff    128 Nov  7  2019 .vscode
drwx------@  78 corinne  staff   2496 Nov 23  2019 Library
-rw-r--r--    1 corinne  staff   1862 Jan  4  2020 .bash_profile-anaconda3.bak
drwx------    4 corinne  staff    128 Jan  6  2020 .config
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 hdf5==1.10.4=hfa1e0ec_0
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 h5py==2.9.0=py27h3134771_0
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 hdf5[version=>=1.10.4,<1.10.5.0a0]
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 libopencv==3.4.2=h0d3447a_1
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 hdf5[version=>=1.10.2,<1.10.3.0a0|>=1.8.18,<1.8.19.0a0|>=1.8.20,<1.9.0a0]
-rw-r--r--    1 corinne  staff      0 Jan  6  2020 hdf5[version=>=1.10.1,<1.10.2.0a0|>=1.10.2,<1.10.3.0a0|>=1.10.4,<1.10.5.0a0|>=1.8.18,<1.8.19.0a0|>=1.8.20,<1.9.0a0]
-rw-------    1 corinne  staff     98 Jan  7  2020 .python_history
drwx------    4 corinne  staff    128 Feb  5  2020 .putty
drwx------    4 corinne  staff    128 Jul  3  2020 .local
drwxr-xr-x  468 corinne  staff  14976 Jul 12  2020 git
-rw-r--r--    1 corinne  staff     53 Jul 12  2020 .gitconfig
-rw-r--r--    1 corinne  staff   2461 Jul 13  2020 .bash_profile
drwx------    5 corinne  staff    160 Jul 13  2020 .ssh
-r--------    1 corinne  staff      9 Aug 25 18:15 .CFUserTextEncoding
drwxr-xr-x    7 corinne  staff    224 Oct  4 18:22 Calibre Library
drwx------@  10 corinne  staff    320 Oct  5 13:09 Documents
drwxr-xr-x    4 corinne  staff    128 Dec 20 10:56 Applications
drwx------+   7 corinne  staff    224 Dec 22 13:15 Movies
drwx------  161 corinne  staff   5152 Jan 11 01:43 .Trash
drwx------+  68 corinne  staff   2176 Jan 12 07:40 Pictures
drwxr-xr-x   17 corinne  staff    544 Jan 12 15:09 getting-started
drwx------@ 185 corinne  staff   5920 Jan 14 17:13 Downloads
drwxr-xr-x    6 corinne  staff    192 Jan 15 01:08 .anaconda_backup
drwxrwsr-x    3 corinne  staff     96 Jan 15 02:26 .conda
-rw-r--r--    1 corinne  staff     17 Jan 15 03:19 .condarc
drwxr-xr-x   15 corinne  staff    480 Jan 15 03:41 miniconda-4.8.3
drwx------@   4 corinne  staff    128 Jan 15 09:58 Creative Cloud Files
drwxr-xr-x   11 corinne  staff    352 Jan 15 09:58 .docker
-rw-r--r--    1 corinne  staff  12288 Jan 15 15:19 .bash_profile.swp
-rw-------    1 corinne  staff  10861 Jan 15 15:19 .viminfo
drwxr-xr-x+  45 corinne  staff   1440 Jan 15 15:19 .
-rw-------    1 corinne  staff  13956 Jan 15 15:19 .bash_history
drwx------   47 corinne  staff   1504 Jan 15 15:19 .bash_sessions
-rw-r--r--@   1 corinne  staff  14340 Jan 15 15:25 .DS_Store
drwx------@ 109 corinne  staff   3488 Jan 15 15:43 Desktop
(base) ChunruideMacBook-Pro:~ corinne$ vi .bash_profile
(base) ChunruideMacBook-Pro:~ corinne$ vim -r .bash_profile

        \export PATH="/Applications/anaconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
# added by Anaconda3 2019.10 installer
# >>> conda init >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$(CONDA_REPORT_ERRORS=false '/Applications/anaconda3/bin/conda' shell.bash hook 2> /dev/null)"
if [ $? -eq 0 ]; then
    \eval "$__conda_setup"
else
    if [ -f "/Applications/anaconda3/etc/profile.d/conda.sh" ]; then
        . "/Applications/anaconda3/etc/profile.d/conda.sh"
        CONDA_CHANGEPS1=false conda activate base
    else
        \export PATH="/Applications/anaconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
alias ll='ls -
```

**Comment by shankari** [comment]
so you need to remove all the anaconda stuff from the `.bash_profile`
you can keep the `alias ll='ls -`
everything else should go

**Comment by corinne-hcr** [comment]
```
# Setting PATH for Python 3.7
# The original version is saved in .bash_profile.pysave
PATH="/Library/Frameworks/Python.framework/Versions/3.7/bin:${PATH}"
export PATH
# added by Anaconda3 2018.12 installer
# >>> conda init >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$(CONDA_REPORT_ERRORS=false '/anaconda3/bin/conda' shell.bash hook 2> /dev/null)"
if [ $? -eq 0 ]; then
    \eval "$__conda_setup"
else
    if [ -f "/anaconda3/etc/profile.d/conda.sh" ]; then
        . "/anaconda3/etc/profile.d/conda.sh"
        CONDA_CHANGEPS1=false conda activate base
    else
        \export PATH="/anaconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
# added by Anaconda3 2019.10 installer
# >>> conda init >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$(CONDA_REPORT_ERRORS=false '/Applications/anaconda3/bin/conda' shell.bash hook 2> /dev/null)"
if [ $? -eq 0 ]; then
    \eval "$__conda_setup"
else
    if [ -f "/Applications/anaconda3/etc/profile.d/conda.sh" ]; then
        . "/Applications/anaconda3/etc/profile.d/conda.sh"
        CONDA_CHANGEPS1=false conda activate base
    else
        \export PATH="/Applications/anaconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
# added by Anaconda3 2019.10 installer
# >>> conda init >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$(CONDA_REPORT_ERRORS=false '/Applications/anaconda3/bin/conda' shell.bash hook 2> /dev/null)"
if [ $? -eq 0 ]; then
    \eval "$__conda_setup"
else
    if [ -f "/Applications/anaconda3/etc/profile.d/conda.sh" ]; then
        . "/Applications/anaconda3/etc/profile.d/conda.sh"
        CONDA_CHANGEPS1=false conda activate base
    else
        \export PATH="/Applications/anaconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
alias ll='ls -
```

**Comment by corinne-hcr** [comment]
> so you need to remove all the anaconda stuff from the .bash_profile
> you can keep the alias ll='ls -
Do you mean I should delete everything from the line ```# added by Anaconda3 2018.12 installer``` to the last third lines?

**Comment by shankari** [comment]
I would remove everything from `<<< conda init <<<` to `# <<< conda init <<<` which is basically everything

**Comment by corinne-hcr** [comment]
> I would remove everything from <<< conda init <<< to # <<< conda init <<< which is basically everything

Do you mean from the sixth line from the beginning to the second line from the bottom?

**Comment by shankari** [comment]
I can't see line numbers on your shell
put in the final `.bash_profile` and I will tell you if it is right

**Comment by corinne-hcr** [comment]
what is the proposed final ```.bash_profile```

**Comment by corinne-hcr** [comment]
Also, since it is in the ```vim -r .bash_profile``` now, should I edit it but not quit? Can I not save and exit safely if my edit is not correct?

**Comment by shankari** [comment]
> what is the proposed final .bash_profile

After all the edits, what will your bash_profile look like?

> Can I not save and exit safely if my edit is not correct?

vim cheatsheet (http://vimsheet.com/) `ESC :q!`

**Comment by corinne-hcr** [comment]
After I edit
```
# Setting PATH for Python 3.7
# The original version is saved in .bash_profile.pysave
PATH="/Library/Frameworks/Python.framework/Versions/3.7/bin:${PATH}"
export PATH
# added by Anaconda3 2018.12 installer

alias ll='ls -
```

**Comment by shankari** [comment]
you can remove the `# added by Anaconda3 2018.12 installer`
Are you using python 3.7? Or can you remove it from the PATH too?

**Comment by corinne-hcr** [comment]
Yes, I am using python 3.7.
Or is there another version of python in the miniconda?

**Comment by corinne-hcr** [comment]
So, should I remove the ```.bash_profile```?

**Comment by shankari** [comment]
your final `bash_profile` should only have the last `alias` line

**Comment by corinne-hcr** [comment]
OK 
After I finishing editing. I use```:wq```to save and quit.
And then I use```vi .bash_profile```. Now it prompts
```
E325: ATTENTION
Found a swap file by the name ".bash_profile.swp"
          owned by: corinne   dated: 五  1 15 15:19:06 2021
         file name: ~corinne/.bash_profile
          modified: YES
         user name: corinne   host name: ChunruideMacBook-Pro.local
        process ID: 1696
While opening file ".bash_profile"
             dated: 五  1 15 20:09:08 2021
      NEWER than swap file!

(1) Another program may be editing the same file.  If this is the case,
    be careful not to end up with two different instances of the same
    file when making changes.  Quit, or continue with caution.
(2) An edit session for this file crashed.
    If this is the case, use ":recover" or "vim -r .bash_profile"
    to recover the changes (see ":help recovery").
    If you did this already, delete the swap file ".bash_profile.swp"
    to avoid this message.

Swap file ".bash_profile.swp" already exists!
[O]pen Read-Only, (E)dit anyway, (R)ecover, (D)elete it, (Q)uit, (A)bort: 
```

**Comment by shankari** [comment]
As I said earlier, you need to delete the `.swp` file manually

**Comment by corinne-hcr** [comment]
After deleting the ```.swp``` file, and the ```.bash_profile``` has the alias line now. What should I do next?

**Comment by shankari** [comment]
From https://github.com/e-mission/e-mission-docs/issues/601#issuecomment-761148400

>  Open a new terminal, check that `which conda` doesn't return anything, or returns the miniconda version only and continue with setup

**Comment by corinne-hcr** [comment]
It seems the ```alias``` causes a problem. Here is what I have in a new terminal.
```
-bash: /Users/corinne/.bash_profile: line 1: unexpected EOF while looking for matching `''
-bash: /Users/corinne/.bash_profile: line 2: syntax error: unexpected end of file

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
ChunruideMacBook-Pro:~ corinne$ which conda
ChunruideMacBook-Pro:~ corinne$ 
```

**Comment by shankari** [comment]
yeah you deleted part of the alias line. The original was probably `alias ll = ls -l` Or if you don't use that alias, you can remove it as well.

**Comment by corinne-hcr** [comment]
OK. It is fine now. ```which conda ```doesn't return anything. 
As you mentioned before, I should check ```.bash_profile``` and ```.bashrc```. But I don't see ```.bashrc```file.
Should I follow the instruction and re-install ```miniconda```now?

**Comment by shankari** [comment]
> Yes, I am using python 3.7.
> Or is there another version of python in the miniconda?

miniconda comes with its own python and you can have different python versions in the different environments

**Comment by shankari** [comment]
If you don't have `.bashrc`, you don't need to check it.
Just follow the instructions now.

And next time, you may want to follow the setup instructions exactly instead of improvising 👍

**Comment by corinne-hcr** [comment]
So, now start from re-install ```miniconda```?

**Comment by shankari** [comment]
yes, run `setup/setup.sh` and follow the instructions (e.g. if it asks you to install miniconda, do so)

**Comment by corinne-hcr** [comment]
Not ```source setup/setup_conda.sh``` from the instruction?

**Comment by shankari** [comment]
if you run `source setup.setup.sh` without running `setup/setup_conda.sh`, it will tell you to setup conda first.
I don't have the installation instructions memorized, just follow the written instructions, or commands in the automated test workflow

**Comment by corinne-hcr** [comment]
Follow this one? ```source setup/setup_conda.sh MacOSX-x86_64```
![image](https://user-images.githubusercontent.com/68228318/104798817-8394e080-577e-11eb-8e58-6b3577809047.png)

**Comment by shankari** [comment]
That is a partial screenshot from a full automated test log.
A link to the full log is above as well.
You should follow all the setup steps from the automated test log and not follow the teardown steps.
The automated test log also verifies the setup, you can verify in the same way

**Comment by corinne-hcr** [comment]
Because above ```Install miniconda``` there are two more steps.
![image](https://user-images.githubusercontent.com/68228318/104798911-90fe9a80-577f-11eb-9f8b-52d03455f4d9.png)
So, should I just start from ```Install miniconda```?

**Comment by shankari** [comment]
> Because above Install miniconda there are two more steps.

That is checkout out the test runner and the repo, which you have already done.
This test focuses on installation, not running, so as you can see from the workflow, we don't start mongo

**Comment by shankari** [comment]
These are basically the same as the installation instructions in the documentation, except that they show the output, so you don't have to worry about whether it is working correctly.

**Comment by corinne-hcr** [comment]
So, basically I can follow the documentation instructions and use the test runner to check if I run correctly in the installation step?

**Comment by shankari** [comment]
sure. If there is a mismatch between the documentation instructions and the test runner, you can ask me. But I don't think there should be anything.

**Comment by corinne-hcr** [comment]
This is what I have after running ```source setup/setup_conda.sh MacOSX-x86_64```

```
ChunruideMacBook-Pro:e-mission-server corinne$ source setup/setup_conda.sh MacOSX-x86_64
Installing for version 4.8.3 and platform MacOSX-x86_64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 53.2M  100 53.2M    0     0  37.1M      0  0:00:01  0:00:01 --:--:-- 60.5M
ERROR: File or directory already exists: '/Users/corinne/miniconda-4.8.3'
If you want to update an existing installation, use the -u option.

     active environment : None
            shell level : 0
       user config file : /Users/corinne/.condarc
 populated config files : /Users/corinne/.condarc
          conda version : 4.8.3
    conda-build version : not installed
         python version : 3.8.3.final.0
       virtual packages : __osx=10.15.7
       base environment : /Users/corinne/miniconda-4.8.3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/corinne/miniconda-4.8.3/pkgs
                          /Users/corinne/.conda/pkgs
       envs directories : /Users/corinne/miniconda-4.8.3/envs
                          /Users/corinne/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.8.3 Darwin/19.6.0 OSX/10.15.7
                UID:GID : 501:20
             netrc file : None
           offline mode : False

# conda environments:
#
                         /Applications/anaconda3
base                  *  /Users/corinne/miniconda-4.8.3

sys.version: 3.8.3 (default, May 19 2020, 13:54:14) 
...
sys.prefix: /Users/corinne/miniconda-4.8.3
sys.executable: /Users/corinne/miniconda-4.8.3/bin/python
conda location: /Users/corinne/miniconda-4.8.3/lib/python3.8/site-packages/conda
conda-build: None
conda-env: /Users/corinne/miniconda-4.8.3/bin/conda-env
user site dirs: ~/.local/lib/python3.7

CIO_TEST: <not set>
CONDA_EXE: /Users/corinne/miniconda-4.8.3/bin/conda
CONDA_PYTHON_EXE: /Users/corinne/miniconda-4.8.3/bin/python
CONDA_ROOT: /Users/corinne/miniconda-4.8.3
CONDA_SHLVL: 0
PATH: /Users/corinne/miniconda-4.8.3/bin:/Users/corinne/miniconda-4.8.3/condabin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin
REQUESTS_CA_BUNDLE: <not set>
SSL_CERT_FILE: <not set>

Successfully installed at /Users/corinne/miniconda-4.8.3. Please activate with 'source setup/activateXXX.sh' in every terminal where you want to use conda
ChunruideMacBook-Pro:e-mission-server corinne$ 
```

As you can see, there is an ```ERROR: File or directory already exists: '/Users/corinne/miniconda-4.8.3'
If you want to update an existing installation, use the -u option.```
I didn't update, but should I deal with this error? Is it successful?
Also, it doesn't show as much information as the test runner does.

Then, I run ```which conda``` , and I get
```
ChunruideMacBook-Pro:e-mission-server corinne$ which conda
/Users/corinne/miniconda-4.8.3/condabin/conda
```
So, is it successful?

**Comment by shankari** [comment]
that certainly looks like it is successful. What additional information does the test runner have?

**Comment by corinne-hcr** [comment]
The additional information looks like the progress of installation and the package information.

There is one thing different from what I have
![image](https://user-images.githubusercontent.com/68228318/104809524-bea90b00-57a2-11eb-8bcd-a81afa212cb3.png)
line 173 & 174, the ```user site dirs ``` from the runner is empty, but from my shell is ```user site dirs: ~/.local/lib/python3.7```
Would it affect my following work?

Can I go to ```Setup the emission environment``` now?

**Comment by shankari** [comment]
>  line 173 & 174, the user site dirs from the runner is empty, but from my shell is user site dirs: ~/.local/lib/python3.7

This is because you had a prior python 3.7 installation. I don't think it should matter, but it is something to keep in mind if we run into problems later.

Yes, you should go on to the next step. I don't think you need to wait for my permission to run every step. I might not be awake!

**Comment by corinne-hcr** [comment]
In the runner, ```Setup the emission environment``` uses different cmd lines (blue lines)
![image](https://user-images.githubusercontent.com/68228318/104809724-18f69b80-57a4-11eb-99e6-f158669bd2a1.png)
while the instruction```Setup the emission environmen``` is ```$ source setup/setup.sh``` and ```$ source setup/setup_notebook.sh```
(I asked about ```$ source setup/setup_notebook.sh``` in the comment above since we need to run ```Timeseries_Sample.ipynb```, and you suggested run this cmd line as well)

At this point, should I follow the documentation instruction instead of the runner?

**Comment by shankari** [comment]
As I said earlier, the runner has additional validation statements - e.g. `conda --version`
The runner also uses different shells ("terminals") for each step, so it might do additional activation steps.
The additional activation steps are not necessary when you are using the same shell ("terminal") but they don't hurt anything.
After validation, and activation, the command is the same (`source setup/setup.sh`)

**Comment by shankari** [comment]
Are you done? I am going to be online for most of today and tomorrow finishing up some work so should be able to respond.

**Comment by corinne-hcr** [comment]
OK. I go to```http://localhost:8080/``` , the browser is fine.

But I have a question. When I setup the ```emission``` environment, I run ```source setup/setup.sh``` and ```source setup/setup_notebook.sh```one after another but I only run ```conda activate emission```after I run ```source setup/setup_notebook.sh```. So it should be fine to use a notebook to open ```Timeseries_Sample.ipynb```, but when I use pycharm to open pther```.py```, can the ```emission```environment work?

**Comment by shankari** [comment]
I don't understand the question. the environment is the set of modules loaded while running. My understanding is that pycharm is an editor, so you would edit files using it. But you would still run from the command line (terminal) right? So as long as you run from a command line using the emission environment, the right set of modules should load.

**Comment by corinne-hcr** [comment]
Because the instruction specifies that

> If you want to use an ipython notebook, use $ source setup/setup_notebook.sh instead

I just wonder if it is still applicable when I don't use a notebook to run other```.py```file

**Comment by corinne-hcr** [comment]
I am going to load data. Before the ```Quick start``` part, there is a link in the instruction and mentions ```docker exec -it "id of running container" bash```
Shoiuld I run it? I think I have connected the server in the previous steps.

**Comment by shankari** [comment]
The instruction has an "if" before it, right? Does the "if" apply to you?

**Comment by corinne-hcr** [comment]
1. Are you talking about this one when you mention "if"? 

> If you want to use an ipython notebook, use $ source setup/setup_notebook.sh instead

When I run ```Timeseries_Sample.ipynb```I should use a notebook, right? But for other```.py```file I don't need to use a notebook. I am not sure if the environment for a notebook also work when I don't use a notebook.

2. I don't know whether I am using docker install. I have run ``` docker run -p 27017:27017 mongo:4.4.0``` and connected the server.

**Comment by shankari** [comment]
> I don't know whether I am using docker install. I have run docker run -p 27017:27017 mongo:4.4.0 and connected the server.

You are not using a docker install. In the two options here: https://github.com/e-mission/e-mission-server/#installation you clicked on the second link, which is clearly labelled manual install. docker install would install the e-mission code into a docker container.

> When I run Timeseries_Sample.ipynbI should use a notebook, right? But for other.pyfile I don't need to use a notebook. I am not sure if the environment for a notebook also work when I don't use a notebook.

As I explained in https://github.com/e-mission/e-mission-docs/issues/601#issuecomment-760515893
only installs 4 new packages. It doesn't remove anything, so the non-notebook environment is still valid.
As you saw from the fact that you were able to start the server.

**Comment by corinne-hcr** [comment]
Then I should probably go to ```Quick start``` and load data.
For loading test data, as I have a terminal connecting the server, should I open a new terminal, go to the e-mission-server directory, activate the ```emission```environment, then run```./e-mission-py.bash bin/debug/load_timeline_for_day_and_user.py emission/tests/data/real_examples/shankari_2015-07-22 test_july_22```?

**Comment by shankari** [comment]
>  For loading test data, as I have a terminal connecting the server, should I open a new terminal, go to the e-mission-server directory, activate the emissionenvironment, then run./e-mission-py.bash bin/debug/load_timeline_for_day_and_user.py emission/tests/data/real_examples/shankari_2015-07-22 test_july_22?

Yes

> Also, every time when I open a new terminal, should I always useconda activate emissionand ./e-mission-py.bash emission/net/api/cfc_webapp.py to start the server? Or just useconda activate emissionto activate the environment before loading data and working on pycharm?

I don't think you need to have the server running at all times, but it doesn't hurt. You do understand that you might write some server code - e.g. an API call from the phone to retrieve trips to label - right?

The server is not some magical entity that is outside your control. The server (at least the analytical parts, and maybe the API layer) are the parts you will be working on. Feel free to use pycharm to edit files, though!

**Comment by corinne-hcr** [comment]
> You do understand that you might write some server code - e.g. an API call from the phone to retrieve trips to label - right?

To be honest, I am not aware of that. Maybe need help on that.

**Comment by shankari** [comment]
yes, when we get there, I will either tell you how to do it, or do it myself depending on where we are in the timeline. As we discussed, the setup took considerably longer than we had originally planned.

**Comment by corinne-hcr** [comment]
When I open a new terminal, and go to the e-mission-server directory, then run ```which conda``` it returns nothing. That's weird.

**Comment by shankari** [comment]
> OK, I see. I should run source setup/activate_conda.sh as well.

Yes! Or you can run `setup/activate.sh` which runs both for you.

for any command line script `--help` will give you the help message. You can use either `-e` or `-u` - they are equivalent beyond the input

**Comment by corinne-hcr** [comment]
Just want to make sure.

1. This is what I have after loading the data
 
> (emission) ChunruideMacBook-Pro:e-mission-server corinne$ ./e-mission-py.bash bin/debug/load_timeline_for_day_and_user.py emission/tests/data/real_examples/shankari_2015-07-22 test_july_22
> storage not configured, falling back to sample, default configuration
> Connecting to database URL localhost
> emission/tests/data/real_examples/shankari_2015-07-22
> Loading file emission/tests/data/real_examples/shankari_2015-07-22
> After registration, test_july_22 -> e8858f39-2d7e-4afd-a0a6-040998c689b0
> Finished loading 0 entries into the usercache and 1906 entries into the timeseries
> 

From the instruction
> This will load the data as a new user with email test_july_22 and a newly created uuid.

does ```uuid```refer to ```e8858f39-2d7e-4afd-a0a6-040998c689b0```?
 
For ```./e-mission-py.bash bin/debug/intake_single_user.py -u <uuid>```, I should replace```<uuid>```with ```e8858f39-2d7e-4afd-a0a6-040998c689b0```, right? 

2. 

> for any command line script --help will give you the help message

So, I should run ```./e-mission-py.bash bin/debug/intake_single_user.py -e --help```?

**Comment by shankari** [comment]
> does uuidrefer to e8858f39-2d7e-4afd-a0a6-040998c689b0?

Yes, assuming that the newly created UUID is `e8858f39-2d7e-4afd-a0a6-040998c689b0`

> So, I should run ./e-mission-py.bash bin/debug/intake_single_user.py -e --help?

That will give you a summary. Have you tried it?

**Comment by shankari** [comment]
After you removed the manually installed anaconda, I haven't seen any major problems from your side.
Can we move back to a model where you work largely on your own and ask me only if you run into problems?

This model of you asking me before you run each and every command is very time-intensive on my part, and is not really faster than my just doing the work myself.

You are a CS student. You showed during your interview that you could figure things out. You can do this!

**Comment by corinne-hcr** [comment]
Yes. I tried two times.
First time

> (emission) ChunruideMacBook-Pro:e-mission-server corinne$ ./e-mission-py.bash bin/debug/intake_single_user.py -e --help
> storage not configured, falling back to sample, default configuration
> Connecting to database URL localhost
> analysis.debug.conf.json not configured, falling back to sample, default configuration
> google maps key not configured, falling back to nominatim
> nominatim not configured either, place decoding must happen on the client
> ERROR:root:habitica not configured, game functions not supported
> Traceback (most recent call last):
>   File "/Users/corinne/Desktop/e-mission/e-mission-server/emission/net/ext_service/habitica/proxy.py", line 22, in <module>
>     key_file = open('conf/net/ext_service/habitica.json')
> FileNotFoundError: [Errno 2] No such file or directory: 'conf/net/ext_service/habitica.json'
> usage: intake_single_user [-h] (-e USER_EMAIL | -u USER_UUID)
> intake_single_user: error: argument -e/--user_email: expected one argument

It looks like I didn't run the right script. 
The second time

> (emission) ChunruideMacBook-Pro:e-mission-server corinne$ ./e-mission-py.bash bin/debug/intake_single_user.py --help
> storage not configured, falling back to sample, default configuration
> Connecting to database URL localhost
> analysis.debug.conf.json not configured, falling back to sample, default configuration
> google maps key not configured, falling back to nominatim
> nominatim not configured either, place decoding must happen on the client
> ERROR:root:habitica not configured, game functions not supported
> Traceback (most recent call last):
>   File "/Users/corinne/Desktop/e-mission/e-mission-server/emission/net/ext_service/habitica/proxy.py", line 22, in <module>
>     key_file = open('conf/net/ext_service/habitica.json')
> FileNotFoundError: [Errno 2] No such file or directory: 'conf/net/ext_service/habitica.json'
> usage: intake_single_user [-h] (-e USER_EMAIL | -u USER_UUID)
> 
> optional arguments:
>   -h, --help            show this help message and exit
>   -e USER_EMAIL, --user_email USER_EMAIL
>   -u USER_UUID, --user_uuid USER_UUID

This shows the meaning of```-e``` and ```-u```

Because I am not familiar with setup, I am not sure if it is correct. I am scared by the experience of the last couple of days.

**Comment by shankari** [comment]
What are you trying to do? If you are trying to find the help message, you succeeded. If you are trying to run the pipeline the help message tells you how. What are you trying to do and what don't you understand?

**Comment by corinne-hcr** [comment]
Oh, I just want to see what```-e``` and ```-u```mean. 

So, after the intake pipeline, I don't need to run the modeling script ```./e-mission-py.bash emission/pipeline/model_stage.py```, right? Am I good to run ```Timeseries_Sample.ipynb``` in an editor as long as I set the editor's environment to ```emission```?

And it is about ```Experimenting with loaded data```. What about ```Running unit tests``` should I run those instructions in the terminal as well?

**Comment by shankari** [comment]
> I am scared by the experience of the last couple of days.

I don't think you need to be. You are done with install. The only thing you can mess up now is in the data install, and you can just purge the data if that happens. Even if you mess up the full install again, `setup/teardown.sh` followed by `setup/setup.sh` will give you a new install.

Don't worry. Have confidence. The goal is for you to learn skills that will help you in college, not to get so paranoid that you can't do anything on your own. You can't ask TAs for this level of help in a 4-year college.

**Comment by corinne-hcr** [comment]
Now I see the problem of ```Timeseries_Sample.ipynb```
![image](https://user-images.githubusercontent.com/68228318/104838184-1d878680-586e-11eb-8795-6afd913b85cf.png)
There is only one row.

So, ```test_user_id = all_users.iloc[1].uuid```will raise ```IndexError```

I also face another problem. When I follow the ```Running unit tests```instruction and run``` ./runAllTests.sh```. It gets stuck on this point.
```FileNotFoundError: [Errno 2] No such file or directory: 'seed_model.json'```

**Comment by shankari** [comment]
> So, is the test successful?

The tests did run successfully.

```
Ran 214 tests in 844.301s

OK
```

> "Running unit tests"

But you don't typically need to run unit tests right now since you haven't made any changes.
https://softwaretestingfundamentals.com/unit-testing/
You will run them later once you have made changes to the code.

>  There is only one row.
> So, test_user_id = all_users.iloc[1].uuidwill raise IndexError

Correct, so how can you fix that?

**Comment by shankari** [comment]
Maybe this wasn't clear to you, but with the end of the installation instructions, the setup is done.
You can now use the system for your needs.
The Quick Start is just that - a suggestion of how to get started. It is expected that you will use those examples to figure out what you will actually do next.

**Comment by shankari** [comment]
And just to clarify once more completely. Your next goal is to figure out how to get the notebook to run fully and show some maps at the end.

**Comment by corinne-hcr** [comment]
I am thinking about loading two more data and there would be three users. 
But after loading ```test_aug_21```(data 2015-aug-21), I run the notebook, the previous one doesn't show.
![image](https://user-images.githubusercontent.com/68228318/104854391-471acf00-58bb-11eb-9bbe-a1d4e13b74a5.png)
The instruction says if I load the same data twice with different user IDs, then only the second one in the database. Here ```user ID``` is the ```user email```, right?
But since I load two different data, why the first one is overwritten?

**Comment by shankari** [comment]
> I am thinking about loading two more data and there would be three users.

That is exactly the thing to do. Did you run the tests between the first load and the second load?
The tests delete everything in the database.

**Comment by corinne-hcr** [comment]
I see.
There are some more problems.

1. 

![image](https://user-images.githubusercontent.com/68228318/104858549-f1ebb700-58d4-11eb-8c9c-30c505c2e62a.png)
Is it an error? But it gives an output. I assume it is fine.

2. 

![image](https://user-images.githubusercontent.com/68228318/104858645-aa195f80-58d5-11eb-8b71-87d38528dc3d.png)

So, I trace back. The first time you call ```ct_df.columns```, there are```start_place``` ```end_place```
![image](https://user-images.githubusercontent.com/68228318/104858880-003ad280-58d7-11eb-88e8-af46122212b2.png)

But after ```# Get cleaned trips for the two users that started on 1st Aug UTC``` and running the ```cs_df = ts.get_data_df("analysis/cleaned_section", time_query=None)```, there are no ```start_place``` ```end_place``` in the columns


![image](https://user-images.githubusercontent.com/68228318/104858949-86efaf80-58d7-11eb-9e0b-839701a827e8.png)

I am not sure if it that is because I didn't load data on ```2017-08-01```and ```2017-08-02```. But there is no such data in the ```real_examples```

3. In the ```real_examples```, some data names have ```.ground_truth```, some don't. What's the difference? For example, ```shankari_2016-08-10```and ```shankari_2016-08-10.ground_truth```

**Comment by shankari** [comment]
>  Is it an error? But it gives an output. I assume it is fine.

It is not an error. It is a deprecation warning. I changed all the server code, but I forgot the notebooks.

> But after # Get cleaned trips for the two users that started on 1st Aug UTC and running the cs_df = 
> ts.get_data_df("analysis/cleaned_section", time_query=None), there are no start_place end_place in the columns

You are conflating two issues.

`ct_df` is a dataframe of confirmed *trips*
`cs_df` is a dataframe of confirmed *sections*

There is no guarantee that they will both have the same columns. You can look at their data models in the links at the top of the notebook.

> Get cleaned trips for the two users that started on 1st Aug UTC

overwrites `ct_df` and as you point out, there are no entries for that date range, so there are no columns.

You can print out `ct_df` *after* the overwrite to confirm.

I think the problem is that I changed the Timeseries_sample to run with the interscity dataset 
https://github.com/interscity/open-data-challenge-2017

At that time, I hosted an online version of the notebook as well, so people could explore the data without installing anything.
Alas, I had to shut down the online version when I left Berkeley.

You can either:
- remove the cell where I am overriding, *or*
- load the interscity dataset using `bin/debug/load_multi_timeline_for_range.py`. You will use clone the repo, and then the `--prefix` option to specify the prefix to the repo. Since you don't need to submit changes to it, you don't need to fork right now.


Since I know you are uncomfortable with running shell scripts, I'm giving you a choice. Either option is OK.

**Comment by corinne-hcr** [comment]
I just comment out the overridden part.

There is only one error left. I am not sure if that matters. But it doesn't affect the map. 
![image](https://user-images.githubusercontent.com/68228318/104865164-1d2fcf80-58f0-11eb-85f6-3411d804de7c.png)

Here is what I have now.
![image](https://user-images.githubusercontent.com/68228318/104865318-7e57a300-58f0-11eb-856a-8926c8f17204.png)

**Comment by corinne-hcr** [comment]
What's the next goal?

**Comment by shankari** [comment]
> There is only one error left. I am not sure if that matters. But it doesn't affect the map.

I'm pretty sure it affects the map because only one trip has loaded. Both `map_list[0]` and `map_list[-1]` are showing the same value.

The next goal is for you to understand the data model before we start working with the data objects. Read through the data model wrappers, experiment with the loaded data, understand the difference between:
- raw and analysed data,
- trips and sections and places

If you want a concrete example to experiment with, you can try to resolve that error. That will also give you experience with reading and understanding existing code, since you will be working with Naomi's code going forward.

If you do plan to try and fix the error, please file a new issue. This is mainly dealing with setup issues at this point.

**Comment by shankari** [comment]
I think that error may be a regression from switching to a new version of folium (in e0c0a2138c0015bd4829c53608b352b531d1b136). I bet that the new version doesn't have a locations attribute for a PolyLine.

Unfortunately, I don't have automated testing for the notebooks, so it is hard to catch small regressions like this.

You could verify that it is a regression by manually uninstalling the current folium from the environment and adding back the older version. If the notebook works in that case, then you can update the code or update the environment.

**Comment by shankari** [comment]
Fixed by #603
